codalab/server/bundle_manager.py

```diff
@@ -378,11 +377,10 @@ def _filter_and_sort_workers(workers_list, bundle, bundle_resources):
         # Filter by tag.
         request_queue = bundle.metadata.request_queue
         if request_queue:
-            tagm = re.match('tag=(.+)', request_queue)
-            if tagm:
-                workers_list = [worker for worker in workers_list if worker['tag'] == tagm.group(1)]
-            else:
-                # We don't know how to handle this type of request queue argument
+            workers_list = self._get_matched_workers(request_queue, workers_list)
+            # Technically this shouldn't happen any more as all the bundles that are specified to run on
+            # invalid private workers are filtered out in _get_staged_bundles_to_run()
+            if not workers_list:
```

Comment:
I'd prefer to be more explicit (`workers_list is None` or `len(workers_list) == 0`)